File: bioskopina_mobile/lib/models/qa.dart
import 'package:json_annotation/json_annotation.dart';

import '../models/qa_category.dart';
import '../models/user.dart';

part 'qa.g.dart';

@JsonSerializable()
class QA {
  int? id;
  int? userId;
  int? categoryId;
  String? question;
  String? answer;
  bool? displayed;
  QAcategory? category;
  User? user;

  QA({
    this.id,
    this.userId,
    this.categoryId,
    this.question,
    this.answer,
    this.displayed,
    this.category,
    this.user,
  });

  factory QA.fromJson(Map<String, dynamic> json) => _$QAFromJson(json);

  Map<String, dynamic> toJson() => _$QAToJson(this);
}

File: backend/Bioskopina.Services/Database/QA.cs
using System;
using System.Collections.Generic;

namespace Bioskopina.Services.Database;

public partial class QA
{
    public int Id { get; set; }

    public int UserId { get; set; }

    public int CategoryId { get; set; }

    public string Question { get; set; } = null!;

    public string Answer { get; set; } = null!;

    public bool Displayed { get; set; }

    public virtual QAcategory Category { get; set; } = null!;

    public virtual User User { get; set; } = null!;
}
